### PR TITLE
feat: Allow user to configure the Tooltip in the Table header

### PR DESCRIPTION
**Pull request by harrisoff**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

Currently there is no way to configure the Tooltip in the Table header.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Allow user to configure the Tooltip in the Table header |
| 🇨🇳 Chinese | 允许用户设置 Table header 中 Tooltip 的属性 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/table/index.en-US.md](https://github.com/harrisoff/ant-design/blob/feature/components/table/index.en-US.md)
[View rendered components/table/index.zh-CN.md](https://github.com/harrisoff/ant-design/blob/feature/components/table/index.zh-CN.md)

### CODE REVIEW

File: components/table/Table.tsx [hunk 1]
```diff
@@ -96,6 +97,7 @@ export interface TableProps<RecordType>
   };
   sortDirections?: SortOrder[];
   showSorterTooltip?: boolean;
```
Comment (afc163) [inline components/table/Table.tsx:99]:
```jsx
showSorterTooltip={{ placement: 'top' }}
```

Comment (harrisoff) [reply components/table/Table.tsx:99]: https://github.com/ant-design/ant-design/blob/master/components/slider/index.tsx#L46

Comment (afc163) [reply components/table/Table.tsx:99]: I believe `showSorterTooltip={{ ... }}` is a better API.

File: components/table/hooks/useSorter.tsx [hunk 2]
```diff
@@ -146,6 +146,15 @@ function injectSorter<RecordType>(
...
       }
+      let tooltipProps: TooltipProps = {
+        title: sortTip,
+      };
+      if (typeof showSorterTooltip === 'object') {
+        tooltipProps = {
+          ...tooltipProps,
+          ...showSorterTooltip,
+        };
+      }
```
Comment (afc163) [inline components/table/hooks/useSorter.tsx:157]:
```suggestion
      const tooltipProps: TooltipProps = typeof showSorterTooltip === 'object' ? showSorterTooltip : {
        title: sortTip,
      };
```

File: components/table/index.zh-CN.md [hunk 3]
```diff
@@ -141,7 +141,7 @@ const columns = [
 | render | 生成复杂数据的渲染函数，参数分别为当前行的值，当前行数据，行索引，@return 里面可以设置表格[行/列合并](#components-table-demo-colspan-rowspan) | function(text, record, index) {} | - |  |
 | responsive | 响应式 breakpoint 配置列表。未设置则始终可见。 | [Breakpoint](https://github.com/ant-design/ant-design/blob/015109b42b85c63146371b4e32b883cf97b088e8/components/_util/responsiveObserve.ts#L1)\[] | - | 4.2.0 |
 | shouldCellUpdate | 自定义单元格渲染时机 | (record, prevRecord) => boolean | - | 4.3.0 |
-| showSorterTooltip | 表头显示下一次排序的 tooltip 提示, 覆盖 table 中 `showSorterTooltip` | boolean | true |  |
+| showSorterTooltip | 表头显示下一次排序的 tooltip 提示, 覆盖 table 中 `showSorterTooltip` | boolean \| [Tooltip props](/components/tooltip/) | true |  |
```
Comment (afc163) [inline components/table/index.zh-CN.md:144]:
```suggestion
| showSorterTooltip | 表头显示下一次排序的 tooltip 提示, 覆盖 table 中 `showSorterTooltip` | boolean \| [Tooltip props](/components/tooltip/#API) | true |  |
```

**Review by afc163 — APPROVED** [review]

File: components/table/index.zh-CN.md [hunk 4]
```diff
@@ -141,7 +141,7 @@ const columns = [
 | render | 生成复杂数据的渲染函数，参数分别为当前行的值，当前行数据，行索引，@return 里面可以设置表格[行/列合并](#components-table-demo-colspan-rowspan) | function(text, record, index) {} | - |  |
 | responsive | 响应式 breakpoint 配置列表。未设置则始终可见。 | [Breakpoint](https://github.com/ant-design/ant-design/blob/015109b42b85c63146371b4e32b883cf97b088e8/components/_util/responsiveObserve.ts#L1)\[] | - | 4.2.0 |
 | shouldCellUpdate | 自定义单元格渲染时机 | (record, prevRecord) => boolean | - | 4.3.0 |
-| showSorterTooltip | 表头显示下一次排序的 tooltip 提示, 覆盖 table 中 `showSorterTooltip` | boolean \| [Tooltip props](/components/tooltip/) | true |  |
+| showSorterTooltip | 表头显示下一次排序的 tooltip 提示, 覆盖 table 中 `showSorterTooltip` | boolean \| [Tooltip props](/components/tooltip/#API) | true |  |
```
Comment (afc163) [inline components/table/index.zh-CN.md:144]: 别忘了英文文档。

Comment (harrisoff) [reply components/table/index.zh-CN.md:144]: ...

**Pull request merged**